src/Symfony/Bundle/FrameworkBundle/Test/KernelTestCaseTrait.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Test;
+
+use PHPUnit\Framework\TestCase;
+
+// Auto-adapt to PHPUnit 8 who added a `void` return-type to the tearDown method
+
+if (method_exists(\ReflectionMethod::class, 'hasReturnType') && (new \ReflectionMethod(TestCase::class, 'tearDown'))->hasReturnType()) {
+eval('
```

Comment:
Do we have to use `eval()` here? Can't we simply declare the trait depending on the PHPUnit version?

Reply:
@xabbuh , I guess we cannot. PHP 5 parser will not recognize `) : void` and the trait will be invalid.